cmake/modules/AddSwift.cmake

```diff
@@ -198,38 +201,69 @@ function(_add_variant_c_compile_flags)
     RESULT_VAR_NAME result)
 
   is_build_type_optimized("${CFLAGS_BUILD_TYPE}" optimized)
-  if(optimized)
-    list(APPEND result "-O2")
 
-    # Add -momit-leaf-frame-pointer on x86.
-    if("${CFLAGS_ARCH}" STREQUAL "i386" OR "${CFLAGS_ARCH}" STREQUAL "x86_64")
-      list(APPEND result "-momit-leaf-frame-pointer")
+  # MSVC has different options to most compilers!
+  if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+    list(APPEND result "-DLLVM_ON_WIN32")
+    list(APPEND result "-D_CRT_SECURE_NO_WARNINGS")
+    # TODO(compnerd) handle /MT
+    list(APPEND result "-D_DLL")
```

Comment:
It doesn't matter if you use "-" or "/" in cl.exe. The problem that I'm mentioning is that the linker gets these flags as well as the compiler, when actually we only want the compiler to have these flags passed to it.